src/test/java/com/philips/research/bombar/core/domain/DependencyTest.java

```diff
@@ -146,6 +146,29 @@ void addsUsages() {
         assertThat(dependencies).containsExactly(target);
     }
 
+    @Test
+    void noStrongUsageForOrphan() {
+        assertThat(dependency.getStrongestUsage()).isEmpty();
+    }
+
+    @Test
+    void findsStrongestRelationShipFromOwnUsage() {
+        final var parent = new Dependency(ID, "Parent");
+        final var firstChild = new Dependency(ID, "First");
+        final var secondChild = new Dependency(ID, "Second");
```

Comment:
This test is now testing two situations:
1. Multiple incoming relationships, where it should pick the strongest.
2. A sibling dependency that should be ignored.

Unit tests should only test one thing at a time. (This makes sure maintainers understand what the test tries to prove, and prevent accidentally passing one of the scenarios due to the more complex context.)

Note that the latter is now already happening: The second child relationship is is ignored because it is weaker than the other two relationships. (Proof: Removing line 159 of Dependency.java does not make this test fail.)

Split the two into separate tests.